config.go

```diff
@@ -428,11 +428,12 @@ func loadConfig() (*config, error) {
 		cfg.Tor.Control, strconv.Itoa(defaultTorControlPort),
 	)
 	switch {
-	case !cfg.Tor.V2 && !cfg.Tor.V3:
-		cfg.Tor.V2 = true
 	case cfg.Tor.V2 && cfg.Tor.V3:
 		return nil, errors.New("either tor.v2 or tor.v3 can be set, " +
 			"but not both")
+	case cfg.DisableListen && (cfg.Tor.V2 || cfg.Tor.V3):
```

Comment:
Should be a check right below this to disable listening (automatically) if tor is set, but v2 and v3 aren't.